pkg/hubble/dropeventemitter/dropeventemitter.go

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package dropeventemitter
+
+import (
+	"context"
+	"slices"
+	"strconv"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	typedv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/record"
+
+	slimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	metaslimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+	slimscheme "github.com/cilium/cilium/pkg/k8s/slim/k8s/client/clientset/versioned/scheme"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	"github.com/cilium/cilium/pkg/identity"
+	client "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+type DropEventEmitter struct {
+	reasons  []string
+	recorder record.EventRecorder
+}
+
+func NewDropEventEmitter(interval time.Duration, reasons []string, k8s client.Clientset) *DropEventEmitter {
+	broadcaster := record.NewBroadcasterWithCorrelatorOptions(record.CorrelatorOptions{
+		BurstSize:            1,
+		QPS:                  1 / float32(interval.Seconds()),
+		MaxEvents:            1,
+		MaxIntervalInSeconds: int(interval.Seconds()),
+		MessageFunc:          func(event *v1.Event) string { return event.Message },
+	})
+	broadcaster.StartRecordingToSink(&typedv1.EventSinkImpl{Interface: k8s.CoreV1().Events("")})
+
+	return &DropEventEmitter{
+		reasons:  reasons,
+		recorder: broadcaster.NewRecorder(slimscheme.Scheme, v1.EventSource{Component: "cilium"}),
+	}
+}
+
+func (e *DropEventEmitter) ProcessFlow(ctx context.Context, flow *flowpb.Flow) error {
+	reason := strings.ToLower(flow.DropReasonDesc.String())
+
+	// Only handle packet drops due to policy related to a Pod
+	if flow.Verdict != flowpb.Verdict_DROPPED ||
+		slices.Contains(e.reasons, reason) ||
```

Comment:
I think the logic should be inverted here, isn't it? If the user specifies they want auth_required and policy_denied, then we should emit events for these drops.
Would you mind extending the unit tests to cover this use-case?

Reply:
Oooh... How did I miss that? Thanks!

I added a test for matching the drop reason.